### PR TITLE
Initial support for Wheatfield Blocked65%

### PR DESCRIPTION
**Pull request by TerryMathews**
Initial support for Wheatfield Blocked65% keyboard.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Initial support for Wheatfield Blocked65%

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/wheatfield/blocked65/blocked65.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	// put your keyboard start-up code here
+	// runs once when the firmware starts up
+	matrix_init_user();
+};
+
+void matrix_scan_kb(void) {
+	// put your looping keyboard code here
+	// runs every cycle (a lot)
+	matrix_scan_user();
+};
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/blocked65.c]: Both these functions can be deleted here; they're the same as the default defined in QMK's core.

https://github.com/qmk/qmk_firmware/blob/4b7a3b2265d476dc3105511c74e25adc13ab8a8a/quantum/matrix_common.c#L21-L23

```suggestion
```

File: keyboards/wheatfield/blocked65/blocked65.h [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include "quantum.h"
+
+// readability
+#define _x_ KC_NO
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/blocked65.h]:
```suggestion
#define LAYOUT_65_ansi_blocker( \
```

File: keyboards/wheatfield/blocked65/blocked65.h [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+){ \
+  { K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E }, \
+  { K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C, K1D, K1E }, \
+  { K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, _x_, K2D, K2E }, \
+  { K30, _x_, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, K3E }, \
+  { K40, K41, K42, _x_, _x_, _x_, K46, _x_, _x_, _x_, K4A, K4B, K4C, K4D, K4E }, \
+}
+
+void matrix_init_user(void);
+void matrix_scan_user(void);
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/blocked65.h]:
```suggestion
```

File: keyboards/wheatfield/blocked65/info.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name": "Wheatfield blocked 65% keyboard", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "bootloader": "",
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/info.json]:
```suggestion
```

File: keyboards/wheatfield/blocked65/info.json [hunk 5]
```diff
@@ -0,0 +1,13 @@
+{
+    "keyboard_name": "Wheatfield blocked 65% keyboard", 
+    "url": "", 
+    "maintainer": "qmk", 
+    "bootloader": "", 
+    "width": 16, 
+    "height": 5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/info.json]:
```suggestion
        "LAYOUT_65_ansi_blocker": {
```

File: keyboards/wheatfield/blocked65/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+// entirely and just use numbers.
+
+enum layers {
+  _BL,
+  _FL
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+  [_BL] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/keymaps/default/keymap.c]:
```suggestion
  [_BL] = LAYOUT_65_ansi_blocker(
```

File: keyboards/wheatfield/blocked65/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+      KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_INSERT,    \
+      KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC, KC_BSLS,  KC_DEL,  \
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,            KC_PGUP,    \
+      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    KC_PGDN,      \
+      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RALT, MO(_FL), KC_LEFT,  KC_DOWN, KC_RIGHT),
+
+  [_FL] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/keymaps/default/keymap.c]:
```suggestion
  [_FL] = LAYOUT_65_ansi_blocker(
```

File: keyboards/wheatfield/blocked65/readme.md [hunk 8]
```diff
@@ -0,0 +1,15 @@
+Wheatfield Blocked65%
+=====
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/readme.md]:
```suggestion
# Wheatfield Blocked65%
```

File: keyboards/wheatfield/blocked65/readme.md [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+
+A 65% PCB sold via TaoBao, commonly sold as part of the Canoe clone referred to as Fanoe.
+
+Keyboard Maintainer: QMK Community<br>
+Hardware Supported: Wheatfield Blocked65%<br>
+Hardware Availability: [taobao.com](https://item.taobao.com/item.htm?id=570827341563&spm=1101.1101.N.N.b124829)<br>
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/readme.md]:
```suggestion
* Keyboard Maintainer: QMK Community
* Hardware Supported: Wheatfield Blocked65%
* Hardware Availability: [taobao.com](https://item.taobao.com/item.htm?id=570827341563&spm=1101.1101.N.N.b124829)
```

File: keyboards/wheatfield/blocked65/rules.mk [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+EXTRAKEY_ENABLE = yes	# Audio control and System control
+CONSOLE_ENABLE = no  	# Console for debug
+COMMAND_ENABLE = no     # Commands for debug and configuration
+NKRO_ENABLE = no		# USB Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+RGBLIGHT_ENABLE = yes   # Enable keyboard underlight functionality
+BACKLIGHT_ENABLE = no  # Enable keyboard backlight functionality
+MIDI_ENABLE = no 		# MIDI controls
+AUDIO_ENABLE = no
+UNICODE_ENABLE = no 		# Unicode
+BLUETOOTH_ENABLE = no # Enable Bluetooth with the Adafruit EZ-Key HID
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/rules.mk]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI controls
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
UNICODE_ENABLE = no         # Unicode

LAYOUTS = 65_ansi_blocker
```

File: keyboards/wheatfield/blocked65/keymaps/default/keymap.c [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+      KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_ENT,            KC_PGUP,    \
+      KC_LSFT,          KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    KC_PGDN,      \
+      KC_LCTL, KC_LGUI, KC_LALT,                          KC_SPC,                 KC_RALT, MO(_FL), KC_LEFT,  KC_DOWN, KC_RIGHT),
+
+  [_FL] = LAYOUT_65_ansi_blocker(
+      KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,   KC_F10,  KC_F11,   KC_F12,  KC_DEL,   KC_PSCR,    \
+      KC_TRNS, KC_NO,   KC_UP,   KC_NO,   RGB_TOG,RGB_VAI,RGB_HUI,RGB_SAI,KC_INS, RESET,   KC_PSCR, KC_SLCK,  KC_PAUS, KC_BSLS,  KC_SLCK, \
+      KC_TRNS, KC_LEFT, KC_DOWN, KC_RIGHT,RGB_MOD,RGB_VAD,RGB_HUD,RGB_SAD,KC_NO,  KC_NO,   KC_F14,  KC_F15,   KC_INS,            KC_HOME,   \
+      KC_LSFT,          KC_MPRV, KC_MPLY, KC_MNXT,KC_NO,  BL_TOGG,KC_NO,  KC_MUTE,KC_VOLD, KC_VOLU, KC_NO,    KC_RSFT, RGB_MOD,  KC_END,       \
+      KC_LCTL, KC_LGUI, KC_LALT,                          RESET,                  KC_RALT, KC_TRNS,           KC_HOME, BL_STEP,  KC_END)
```
Comment (noroadsleft) [inline keyboards/wheatfield/blocked65/keymaps/default/keymap.c]: Something a bit cleaner and more organized.

```suggestion
    [_BL] = LAYOUT_65_ansi_blocker(
        KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_INS,
        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS, KC_DEL,
        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,  KC_PGUP,
        KC_LSFT,          KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN,
        KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                             KC_RALT, MO(_FL), KC_LEFT, KC_DOWN, KC_RGHT
    ),

    [_FL] = LAYOUT_65_ansi_blocker(
        KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  KC_PSCR,
        KC_TRNS, KC_NO,   KC_UP,   KC_NO,   RGB_TOG, RGB_VAI, RGB_HUI, RGB_SAI, KC_INS,  RESET,   KC_PSCR, KC_SLCK, KC_PAUS, KC_BSLS, KC_SLCK,
        KC_TRNS, KC_LEFT, KC_DOWN, KC_RGHT, RGB_MOD, RGB_VAD, RGB_HUD, RGB_SAD, KC_NO,   KC_NO,   KC_F14,  KC_F15,           KC_INS,  KC_HOME,
        KC_LSFT,          KC_MPRV, KC_MPLY, KC_MNXT, KC_NO,   BL_TOGG, KC_NO,   KC_MUTE, KC_VOLD, KC_VOLU, KC_NO,   KC_RSFT, RGB_MOD, KC_END,
        KC_LCTL, KC_LGUI, KC_LALT,                            RESET,                              KC_RALT, KC_TRNS, KC_HOME, BL_STEP, KC_END
    )

```

**Comment by TerryMathews** [comment]
The board they are shipping now is hotswap and supports only one layout.

**Comment by TerryMathews** [comment]
Sorry, was closed in error. Is there anything else that needs to be done before merge?

**Comment by noroadsleft** [comment]
> Sorry, was closed in error. Is there anything else that needs to be done before merge?

No, looks good, thanks!

**Pull request merged**